docs/user_guides/debugging/stepping.rst

```diff
@@ -40,58 +27,162 @@ Run debugger:
+    (gdb) step
+    __main__::func_sum () at simple_dppy_func.py:22
+    22          result = a_in_func + b_in_func
+    (gdb) continue
+    Continuing.
+
+``stepi``
+--------
+
+The command allows you to move forward in machine instructions. The example uses an additional command ``x/i $pc``, which print the instruction to be executed.
+
+.. code-block:: bash
+
+    export NUMBA_DPPY_DEBUGINFO=1
+    export NUMBA_OPT=1
+    gdb-oneapi -q --args python simple_sum.py
+
+GDB output:
+
+.. code-block:: bash
+
+    (gdb) break simple_sum.py:22
+    Breakpoint 1 (simple_sum.py:22) pending.
+    (gdb) run
+    ...
+    Thread 2.2 hit Breakpoint 1, with SIMD lanes [0-7], __main__::data_parallel_sum () at simple_sum.py:22
+    22          i = dppy.get_global_id(0)
+    (gdb) display/i $pc
+    1: x/i $pc
+    => 0xfffeb5d0 <_ZN8__main__17data_parallel_sumE9DPPYArrayIfLi1E1C7mutable7alignedE9DPPYArrayIfLi1E1C7mutable7alignedE9DPPYArrayIfLi1E1C7mutable7alignedE+1488>:
+        (W)     mov (1|M0)               r53.1<1>:w    0:w                              
+    (gdb) stepi
+    0x00000000fffeb5e0      22          i = dppy.get_global_id(0)
+    1: x/i $pc
+    => 0xfffeb5e0 <_ZN8__main__17data_parallel_sumE9DPPYArrayIfLi1E1C7mutable7alignedE9DPPYArrayIfLi1E1C7mutable7alignedE9DPPYArrayIfLi1E1C7mutable7alignedE+1504>: (W)     cmp (1|M0)    (eq)f0.0   null<1>:w     r53.1<0;1,0>:w    0:w              
+    (gdb) stepi
+    [Switching to Thread 1.1073742080 lane 0]
+
+    Thread 2.3 hit Breakpoint 1, with SIMD lanes [0-1], __main__::data_parallel_sum () at simple_sum.py:22
+    22          i = dppy.get_global_id(0)
+    1: x/i $pc
+    => 0xfffeb5d0 <_ZN8__main__17data_parallel_sumE9DPPYArrayIfLi1E1C7mutable7alignedE9DPPYArrayIfLi1E1C7mutable7alignedE9DPPYArrayIfLi1E1C7mutable7alignedE+1488>: (W)     mov (1|M0)               r53.1<1>:w    0:w                              
+    (gdb) continue
+    Continuing.
+    Done...
 
 ``next``
 --------
 
-Stepping-like behavior.
+Stepping-like behavior, but the command does not go into nested functions.
+
+.. code-block:: bash
+
+    export NUMBA_DPPY_DEBUGINFO=1
+    export NUMBA_OPT=1
+    gdb-oneapi -q --args python simple_dppy_func.py
+
+GDB output:
+
+.. code-block:: bash
+
+    (gdb) break simple_dppy_func.py:28
+    Breakpoint 1 (simple_dppy_func.py:28) pending.
+    (gdb) run
+    ...
+    Thread 2.2 hit Breakpoint 1, with SIMD lanes [0-7], __main__::kernel_sum () at simple_dppy_func.py:28
+    28          i = dppy.get_global_id(0)
+    (gdb) next
+    Thread 2.3 hit Breakpoint 1, with SIMD lanes [0-1], __main__::kernel_sum () at simple_dppy_func.py:28
+    28          i = dppy.get_global_id(0)
+    (gdb) next
+    29          c_in_kernel[i] = func_sum(a_in_kernel[i], b_in_kernel[i])
+    (gdb) next
+    Done...
+
+.. _single_stepping:
+
+``set scheduler-locking step``
+-------------------------------
+
+Debug of the first line of the kernel and functions works out twice.
```

Comment:
Please, replace it with explanation from documentation.

Reply:
The documentation uses an example, so I only took the explanation from it.
Full info with example  is in see also link